### ISSUE TITLE
What should creating xrCompatible GL context do if there is no active immersive device?

### ISSUE THREAD

**Issue by asajeffrey** [immersive-web/webxr#778]
The spec text for `makeXrCompatible()` (https://immersive-web.github.io/webxr/#dom-webglrenderingcontextbase-makexrcompatible) and for `getContext({ xrCompatible: true })` refer to "the immersive XR device" (https://immersive-web.github.io/webxr/#xr-immersive-xr-device) which is the active immersive XR device. It's not obvious what is expected to happen if no device is currently active.

This has the possibility to be a footgun, for example the code:
```js
navigator.xr.requestSession("immersive-vr").then((session) => {
  let glCanvas = document.createElement("canvas");
  let gl = glCanvas.getContext("webgl", { xrCompatible: true });
  ...
});
```
is fine because there's an active immersive device when the GL context is created, but the code:
```js
let glCanvas = document.createElement("canvas");
let gl = glCanvas.getContext("webgl", { xrCompatible: true });
navigator.xr.requestSession("immersive-vr").then((session) => {
  ...
});
```
is not.

If `getContext` took the session object as a parameter, it would be more obvious what the indended semantics was, e.g. something like:
```js
navigator.xr.requestSession("immersive-vr").then((session) => {
  let glCanvas = document.createElement("canvas");
  let gl = glCanvas.getContext("webgl", { xrCompatible: session });
  ...
});
```

**Comment by toji** [comment]
There's a small but understandable misreading of the spec that accounts for some confusion here. The [immersive XR device](https://immersive-web.github.io/webxr/#xr-immersive-xr-device) is the hardware that the UA has designated the API will interact with, regardless of whether or not there is an active **session**. The immersive device may be `null` if there is no appropriate hardware available, and in that case making the context XR compatible should be a no-op. But if new hardware is connected after that point the context would need to call `makeXRCompatible()` in order to be used with any future immersive sessions created. (This is indicated by the `devicechanged` event firing on `navigator.xr`).

I do see two opportunities to improve the spec text in this regard:
 1) We should specify that the algorithms that make the context compatible should [ensure an immersive device is selected](https://immersive-web.github.io/webxr/#ensures-an-immersive-xr-device-is-selected) before proceeding.
 2) We should indicate that if the immersive xr device is `null` after ensuring one is selected the algorithm can set the XR compatible flag to `false` early-terminate.

**Comment by asajeffrey** [comment]
Ah, part of the problem here is that the word "active" at https://github.com/servo/servo/blob/196c511d5ebf81b3fe202c5e0c5c1972a6408ab7/components/constellation/event_loop.rs#L21 is not defined, and the nearest definition is for "active immersive session" https://immersive-web.github.io/webxr/#active-immersive-session but apparently that's not what's meant here.

**Comment by Manishearth** [comment]
Fixed in #809
